tests/test_visualizer.py

```diff
@@ -28,7 +28,8 @@ def perturb(input, target, model):
     adversary = Mock(spec=Adversary, side_effect=perturb)
 
     visualizer = PerturbedImageVisualizer(folder)
-    visualizer.on_run_end(adversary, input_list, target_list, model)
+    kwargs = {"input": input_list, "target": target_list, "model": model}
+    visualizer.on_run_end(adversary, **kwargs)
```

Comment:
Get rid of `kwargs` and just call the function normally with keyword arguments.

Reply:
Fixed.